tools/nl/nodejs_query_differ/README.md

```diff
@@ -11,4 +11,28 @@ To run this:
 ```
 
 - ENV is the environment that the NodeJs Query tests were run in. Default is `autopush`
-- BASE_FOLDER and TEST_FOLDER are the dates in the folder path of the folders containing the results of the NodeJs Query runs to test and to test against. All NodeJs Query test results are saved in [gs://datcom-website-periodic-testing](https://pantheon.corp.google.com/storage/browser/datcom-website-periodic-testing) and the folder path has the form `/<env>/<date>/nodejs_query/`. Default for BASE_FOLDER is the date of the 2nd last run and default for TEST_FOLDER is the date of the last run.
\ No newline at end of file
+- BASE_FOLDER and TEST_FOLDER are the dates in the folder path of the folders containing the results of the NodeJs Query runs to test and to test against. All NodeJs Query test results are saved in [gs://datcom-website-periodic-testing](https://pantheon.corp.google.com/storage/browser/datcom-website-periodic-testing) and the folder path has the form `/<env>/<date>/nodejs_query/`. Default for BASE_FOLDER is the date of the 2nd last run and default for TEST_FOLDER is the date of the last run.
+
+
+## Update goldens
+
+### Fetch goldens
+
+To get new goldens and save it locally (to easily see the diffs between the new and old goldens), you can run:
+
+```bash
+./update_goldens.sh -m fetch -e <ENV> [-g <GOLDEN_FOLDER>]
+```
+
+- ENV is the environment to update the goldens for. Default is 'autopush'
+- GOLDEN_FOLDER is the date in the folder path of the folders containing the results of the NodeJs Query runs to use the results as the golden set. All NodeJs Query test results are saved in [gs://datcom-website-periodic-testing](https://pantheon.corp.google.com/storage/browser/datcom-website-periodic-testing) and the folder path has the form `/<env>/<date>/nodejs_query/`. Default for GOLDEN_FOLDER is the date of the last run.
+
+### Push goldens
+
+To push the new goldens that have been saved locally to gcs (The differ runs diffs against the gcs goldens), you can run:
+
+```bash
+./update_goldens.sh -m push -e <ENV>
+```
+
+- ENV is the environment to update the goldens for. Default is 'autopush'
```

Comment:
Just so I understand, is there a reason the test doesn't read the jsons from the checked-in version?

Reply:
I chose to put it in gcs so that if we want to update the goldens, we don't have to go through an extra step of rebuilding the cron testing job docker image to use these new goldens.. but I guess we could also update cloudbuild to re-build cron testing job docker image on PR submissions. Do you have a preference

Reply:
Having this checked-in and auto-building docker image would be very nice.  But I'm not sure how much more work the latter is?

Reply:
auto-building docker image should be simple, but will need to test out packaging the golden files with the docker image and then using that to run the differ. Let me try it out!